### PR TITLE
Use PSR-6 for accessing the query cache

### PR DESCRIPTION
**Pull request by derrabus**

### CODE REVIEW

**Comment by greg0ire** [comment]
Converting to draft since there is a `TODO`

**Comment by derrabus** [comment]
> Converting to draft since there is a `TODO`

🙈 

Fixed.

File: lib/Doctrine/ORM/Query.php [hunk 1]
```diff
@@ -489,16 +514,21 @@ public function useQueryCache($bool): self
...
+            'https://github.com/doctrine/orm/pull/9004',
+            '%s is deprecated and will be removed in Doctrine 3.0 without replacement.',
+            __METHOD__
+        );
 
-        $queryCache = $this->_em->getConfiguration()->getQueryCache();
+        $queryCache = $this->queryCache ?? $this->_em->getConfiguration()->getQueryCache();
 
         return $queryCache ? DoctrineProvider::wrap($queryCache) : null;
     }
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Query.php:534]: Let's take this as an occasion to add tests for this method? The added lines result in a drop in coverage.

Comment (derrabus) [reply lib/Doctrine/ORM/Query.php:534]: #9007

**Comment by derrabus** [comment]
I'm rebasing to make sure the new test is actually run.

**Review by greg0ire — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Hi @derrabus @greg0ire,

Priori to this PR I use
```
$query = $this->createQueryBuilder('foo')->(...)->getQuery();

if ($invalidateCache) {
     $query->getQueryCacheDriver()?->delete(self::CACHE_KEY);
}

return $query->enableResultCache(self::CACHE_LIFETIME, self::CACHE_KEY)->getOneOrNullResult();
```

What would be the new way to achieve the same behavior now `getQueryCacheDriver` is deprecated ?

I'd like to be able to delete a specific cache entry, and this PR doesn't introduce a `getQueryCache` method.

**Comment by greg0ire** [comment]
`$configuration->getQueryCache()` would be the way I suppose.

**Comment by VincentLanglet** [comment]
> `$configuration->getQueryCache()` would be the way I suppose.

Thanks, I can indeed solve it with
```
$queryCache = $this->getEntityManager()->getConfiguration()->getQueryCache();
$queryCache?->deleteItem(self::CACHE_KEY);
```

**Pull request merged**